### PR TITLE
Update Python Builder to use latest git for compatibility with github actions

### PR DESCRIPTION
**Pull request by axsaucedo**
Currently the Python lint test is failing due to old version of git being used (2.11) as per https://github.com/actions/checkout/issues/126

This PR upgrades the version of git, which is ultimately required in the `python lint` job as it performs a diff with master.

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **axsaucedo**
You can assign the PR to them by writing `/assign @axsaucedo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by axsaucedo** [comment]
@RafalSkolasinski are there any further changes required?

**Comment by RafalSkolasinski** [comment]
As long as this fixes the problem it looks all good and reasonable to me

**Review by RafalSkolasinski — APPROVED** [review]
/lgtm

**Pull request merged**
